api/src/main/java/io/opentelemetry/api/baggage/EntryMetadata.java

```diff
@@ -5,34 +5,63 @@
 
 package io.opentelemetry.api.baggage;
 
-import com.google.auto.value.AutoValue;
 import javax.annotation.concurrent.Immutable;
 
 /**
  * {@link EntryMetadata} contains properties associated with an {@link Entry}. This is an opaque
  * wrapper for a String metadata value.
  */
 @Immutable
-@AutoValue
-public abstract class EntryMetadata {
-  public static final EntryMetadata EMPTY = create("");
+public final class EntryMetadata {
+  private static final EntryMetadata EMPTY = create("");
 
-  EntryMetadata() {}
+  /** Returns an {@link EntryMetadata} with no value. */
+  public static EntryMetadata empty() {
+    return EMPTY;
+  }
 
-  /**
-   * Creates an {@link EntryMetadata} with the given value.
-   *
-   * @param metadata TTL of an {@code Entry}.
-   * @return an {@code EntryMetadata}.
-   */
+  /** Returns an {@link EntryMetadata} with the given value. */
   public static EntryMetadata create(String metadata) {
-    return new AutoValue_EntryMetadata(metadata);
+    if (metadata == null) {
+      return empty();
+    }
+    return new EntryMetadata(metadata);
+  }
+
+  private final String metadata;
+
+  private EntryMetadata(String metadata) {
+    this.metadata = metadata;
   }
 
   /**
    * Returns the String value of this {@link EntryMetadata}.
    *
    * @return the raw metadata value.
    */
-  public abstract String getValue();
+  public String getValue() {
+    return metadata;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof EntryMetadata)) {
+      return false;
+    }
+    EntryMetadata that = (EntryMetadata) o;
+    return metadata.equals(that.metadata);
+  }
+
+  @Override
+  public int hashCode() {
+    return metadata.hashCode();
+  }
+
+  @Override
+  public String toString() {
+    return "'" + metadata + "'";
```

Comment:
Changed this a bit, I think the previous autovalue implementation would have been `EntryMetadata{metadata=foo}`

Reply:
I think the old version was a bit better. Any time we make a `toString()` look too usable by an end-user for something other than debugging, I think we open ourselves up to people mis-using it, which would lock us into a particular implementation.